smac_planner/src/a_star.cpp

```diff
@@ -417,16 +417,35 @@ void AStarAlgorithm<NodeT>::addNode(const float cost, NodePtr & node)
 // Might need a new NodeT::getTravelCost() so each can have their own to get cost
 // TODO g cost include change direction + forward / back penalties.
 // https://github.com/karlkurzer/path_planner/blob/master/src/node3d.cpp#L73-L102
+template<>
+float AStarAlgorithm<NodeSE2>::getTraversalCost(
```

Comment:
In my SE2 collision PR, I moved getTraversalCost into the Node templates themselves, so that you can make the specific logic. We shouldn't override the template for something this low into A* (at that point, it wouldn't be a templated A* as much as a bunch of implementations of functions overriding templates). 

This is also generally unnecessary. You're multiplying by a constant, which will have no affect on expansion, the order will be the exact same. Each motion model moves the exact same `d`.

Reply:
Though I think if we remove the `d`, we can just revert to where it was before since it would be identical.

Reply:
The reason for multiplying by `d` was that the actual cost was being calculated as less than the heuristic. This helped to stop that happening.
I'm good with moving the cost into the Node. I will wait for your PR.

Reply:
Offending commit removed.